x-pack/plugins/fleet/server/constants/fleet_es_assets.ts

```diff
@@ -92,21 +92,28 @@ _meta:
 description: >
   Final pipeline for processing all incoming Fleet Agent documents.
 processors:
-  - date:
+  - script:
       description: Add time when event was ingested (and remove sub-seconds to improve storage efficiency)
       tag: truncate-subseconds-event-ingested
-      field: _ingest.timestamp
-      target_field: event.ingested
-      formats:
-        - ISO8601
-      output_format: date_time_no_millis
       ignore_failure: true
+      source: |-
+        if (ctx?.event == null) {
+          ctx.event = [:];
+        }
+
+        ctx.event.ingested = metadata().now.withNano(0).format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
   - remove:
       description: Remove any pre-existing untrusted values.
       field:
         - event.agent_id_status
         - _security
       ignore_missing: true
+  - remove:
+      description: Remove event.original unless the preserve_original_event tag is set
+      field: event.original
+      if: "ctx?.tags == null || !(ctx.tags.contains('preserve_original_event'))"
+      ignore_failure: true
+      ignore_missing: true
```

Comment:
@joshdover @ishleenk17 - I've added the `remove` processor for `event.original` here, but I'm not sure if this is working as intended. What would be a good way to manually test this? I tried adding `preserve_original_event` tags to a bunch of system datastreams but I don't see `event.original` coming through in the resulting documents. Maybe I'm missing something?

Reply:
Understood. The automated tests I added do seem to be working, so I'm fine to ship this without a great sense of how to manually test if it's deemed relatively harmless.